File: c++/0231.cpp
// Link: https://leetcode.com/problems/power-of-two/description/
// Runtime: 0 ms
// Memory: 7.13 MB

class Solution 
{
public:
    bool isPowerOfTwo(int n) 
    {
        if (n < 1)
            return false;

        return std::fabs(std::log2(n) - static_cast<int>(std::log2(n))) < std::numeric_limits<double>::epsilon();
    }
};

File: python/0231.py
# Link: https://leetcode.com/problems/power-of-two/description/
# Runtime: 29 ms
# Memory: 16.57 MB


class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n < 1:
            return False

        return math.fabs(math.log2(n) - int(math.log2(n))) < sys.float_info.epsilon
